### DISCUSSION TITLE
How do I implement some health system in ink?

### DISCUSSION THREAD

**Discussion by fireton** [inkle/ink#729] [Writing in ink]
Say I have `health` variable and when it become zero I divert to say special «death» knot.

I see I can't divert from functions so how to implement health check at every move?

To put some `->check_health->` tunnel at every knot and stitch would be very cumbersome...

**Comment by GreenCloversGames** [comment]
Functions are special versions of knots, so they only need to be used if they will return a value. If you're just checking a variable, you can create a tunnel to take damage and use this every time you want to change the health variable. 




```
VAR health = 100

->Main

=Main

I have {health} health. 

+[Take 50 damage]
    Oh dear, I have taken 50 damage.
    ->take_damage(50) -> Main

+[Take 20 damage]
    Oh dear, I have taken 20 damage.
    ->take_damage(20) -> Main


->END

=take_damage(damage)
    ~ health -= damage
    -> check_dead ->
    ->->

=check_dead
    {health <= 0:
        Oh no, I'm dead.
        ->END
    }
    ->->
```
